packages/store-indexer/src/compress.ts

```diff
@@ -0,0 +1,48 @@
+import { Middleware } from "koa";
+import { Readable, Stream } from "node:stream";
+import accepts from "accepts";
+import { Zlib, createBrotliCompress, createDeflate, createGzip } from "node:zlib";
+import { includes } from "@latticexyz/common/utils";
+
+// Loosely based on https://github.com/holic/koa-compress/blob/master/lib/index.js
+// with better handling of streams better with occasional flushing
+
+const encodings = {
+  br: createBrotliCompress,
+  gzip: createGzip,
+  deflate: createDeflate,
+} as const;
+
+const encodingNames = Object.keys(encodings) as (keyof typeof encodings)[];
+
+function flushEvery<stream extends Zlib & Readable>(stream: stream, bytesThreshold: number): stream {
+  let bytesSinceFlush = 0;
+  stream.on("data", (data) => {
+    bytesSinceFlush += data.length;
+    if (bytesSinceFlush > bytesThreshold) {
+      bytesSinceFlush = 0;
+      stream.flush();
+    }
+  });
+  return stream;
```

Comment:
Would it be possible that the "last" bytes are stuck in the stream if the data isn't a multiple of `bytesThreshold`? Would we need some kind of timeout after which we flush even if there is no more data event?

Reply:
> Would it be possible that the "last" bytes are stuck in the stream if the data isn't a multiple of `bytesThreshold`? Would we need some kind of timeout after which we flush even if there is no more data event?

This might be possible for an open and long running stream, yes. But shouldn't matter as much while we're closing the stream when we finish streaming data down.

The stream is flushed when the stream closes (which is separate from `compressed.end(ctx.body)`, which pushes+closes the entirety of the body to the compressed, for non-streaming bodies).

I think it would be best to expose a `.flush` method in this middleware that we can access downstream in the `eventStream` middleware, but it was tricky to get it right with types, so I punted on that for now.